lib/awful/tooltip.lua

```diff
@@ -621,6 +627,10 @@ function tooltip.new(args)
     self._private.align   = args.align or beautiful.tooltip_align  or "right"
     self._private.shape   = args.shape or beautiful.tooltip_shape
                                 or shape.rectangle
+    self._private.tooltip_gaps  = args.tooltip_gaps or beautiful.tooltip_gaps or {
+        left = args.tooltip_gaps or 0, right  = args.tooltip_gaps or 0,
+        top  = args.tooltip_gaps or 0, bottom = args.tooltip_gaps or 0
+    }
```

Comment:
> **args.tooltip_gaps** or beautiful.tooltip_gaps or { left = **args.tooltip_gaps** or 0, ... }

i can't undeerstand what do you mean with that

Reply:
@actionless is there a better way to set the margins? I just copied the table from awful.placement https://github.com/awesomeWM/awesome/blob/f997009ff4c4cb73c1364126e6df7867832705d3/lib/awful/placement.lua#L317-L320

Reply:
@Elv13 this still looks like it should be fixed before merging it

Reply:
It was merged, but this looks fine to me

Reply:
it could be safely simplified to just ` = args.tooltip_gaps or beautiful.tooltip_gaps or 0`

Reply:
Yeah, I think it should be `= args.tooltip_gaps or beautiful.tooltip_gaps or 0` because awful.placement already handles the table part. Also, I forgot to add

```lua
self:emit_signal("property::gaps", val)
if self._private.visible then
    set_geometry(self)
end
```

in the set_gaps(). Sorry for the mess. :grimacing: